ipemu/src/TestBench.scala

```diff
@@ -5,19 +5,70 @@ package org.chipsalliance.t1.ipemu
+  val dpiDumpWave = Module(new DpiDumpWave)
+  val dpiFinish = Module(new DpiFinish)
+  val dpiError = Module(new DpiError)
+  val dpiInit = Module(new DpiInitCosim)
+  val dpiTimeoutCheck = Module(new TimeoutCheck(TimeoutCheckParameter(clockRate)))
+
+  val clock: Clock = clockGen.clock.asClock
+  val reset: Bool = clockGen.reset
+
   val dut: T1 = withClockAndReset(clock, reset)(Module(generator.module()))
 
-  val verificationModule = Module(new VerificationModule(dut))
-  dut.request <> verificationModule.req
-  dut.response <> verificationModule.resp
-  dut.csrInterface <> verificationModule.csrInterface
-  dut.storeBufferClear <> verificationModule.storeBufferClear
-  dut.memoryPorts <> verificationModule.tlPort
-  clock := verificationModule.clock
-  reset := verificationModule.reset
- }
+  // Driver to Scalar Core
+  val pokeInst = Module(new PokeInst(PokeInstParameter(dut.parameter.xLen, dut.parameter.laneParam.vlMaxBits, latPokeInst)))
+  val tlPoke = Seq.fill(dut.parameter.lsuBankParameters.size)(Module(new PokeTL(dut.parameter.tlParam.bundle(), latPokeTL)))
+
+  // DPI Monitors
+  val tlPeek = Seq.fill(dut.parameter.lsuBankParameters.size)(Module(new PeekTL(dut.parameter.tlParam.bundle(), latPeekTL)))
+
+  // Monitors
+  // TODO: These monitors should be purged out after offline difftest is landed
+  val peekIssue = Module(new PeekIssue(PeekIssueParameter(dut.parameter.instructionIndexBits, latPeekIssue)))
+  val peekLsuEnq = Module(new PeekLsuEnq(PeekLsuEnqParameter(dut.parameter.lsuParameters.lsuMSHRSize, latPeekLsuEnq)))
+  val peekWriteQueue = Seq.tabulate(dut.parameter.laneNumber)( laneIdx => Module(new PeekWriteQueue(PeekWriteQueueParameter(
+    dut.parameter.vrfParam.regNumBits,
+    dut.parameter.laneNumber,
+    dut.parameter.vrfParam.vrfOffsetBits,
+    dut.parameter.vrfParam.instructionIndexBits,
+    dut.parameter.vrfParam.datapathWidth,
+    negLatPeekWriteQueue,
+  ))))
+  val peekVRFWriteQueue = Seq.tabulate(dut.parameter.laneNumber)( laneIdx => Module(new PeekVrfWrite(PeekVrfWriteParameter(
+    dut.parameter.vrfParam.regNumBits,
+    dut.parameter.laneNumber,
+    dut.parameter.vrfParam.vrfOffsetBits,
+    dut.parameter.vrfParam.instructionIndexBits,
+    dut.parameter.vrfParam.datapathWidth,
+    negLatPeekWriteQueue,
+  ))))
+
+  // TODO: connect signals based on [[VerificationModule]],
+  //       original VerificationModule use boring utils to connect signals to DPI modules
+  //       in this PR, we should manually expose those verification-only APIs via probe, and connect to T1.
+  //       It can keep API stable and documented.
+  // It should read signals from probe as a new signal here.
+  dut.probeStage0EnqueueReady.zipWithIndex.flatMap(lane =>
+    lane._1.zipWithIndex.map { slot =>
+      val data = Wire(chiselTypeOf(slot._1)).suggestName(s"Lane${lane._2}Slot${slot._2}ProbeStage0EnqueueReady")
+      data := probe.read(slot._1)
+      data
+    }
+  )
```

Comment:
Use API like this to read signals from Probe IO.
For signals sending to `tlPeek`, `peekIssue`, and other DPI signals.
read it from `dut` and send to via `:=`